docs/sts/tls.md

```diff
@@ -0,0 +1,126 @@
+# AssumeRoleWithCertificate [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io)
+
+## Introduction
+
+MinIO provides a custom STS API that allows authentication with client X.509 / TLS certificates.
+
+A major advantage of certificate-based authentication compared to other STS authentication methods,
+like OpenID Connect or LDAP/AD, is that client authentication works without any additional/external
+component that must be constantly available. Therefore, certificate-based authentication may provide
+better availability / lower operational complexity.
+
+The MinIO TLS STS API can be configured via MinIO's standard configuration API (i.e. using `mc admin config set/get`).
+Further, it can be configured via the following environment variables:
+
+```
+mc admin config set myminio identity_tls --env
+KEY:
+identity_tls  enable X.509 TLS certificate SSO support
+
+ARGS:
+MINIO_IDENTITY_TLS_STS_EXPIRY   (duration)  temporary credentials validity duration in s,m,h,d. Default is "1h"
```

Comment:
To keep it consistent with other STS APIs, we should use a "DurationSeconds" query parameter in the API to set per-call expiry, and remove this environment configuration. We are also using an upper limit of 1 year validity for all STS creds.